### PR TITLE
Add stages to aggregation pipeline

### PR DESCRIPTION
**Pull request by herberthk**
### Updated aggregation pipeline
- [x]  $collstats to the aggregation pipeline and added assertions
- [x]  $limit to aggregation pipeline and added assertions
- [x]   $skip to the aggregation pipeline and added assertions
- [x]   $geoNear to the aggregation pipeline and added assertions

### CODE REVIEW

**Comment by herberthk** [comment]
@tianhuil allow contributors to be able to update the label of the PR, thanks in advance

**Comment by herberthk** [comment]
Also, the example links on the main repository you provided [this link](https://github.com/tianhuil/aerial-app/pull/804) and [this link](https://github.com/tianhuil/aerial-app/pull/805) are not accessible

File: src/types/geoNear.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (tianhuil) [inline src/types/geoNear.ts:1]: These are actually already defined already.  Use those definitions instead (or update them if you don't think they're correct).

File: src/types/aggregation.assert.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    count: {},
+    queryExecStats: {}
+}
+
+type IncorrectCollStatsExample = {
+    latencyStats: { histograms: number },
+    storageStats: { scale: string },
+    count: {},
+    queryExecStats: {}
+}
```
Comment (tianhuil) [inline src/types/aggregation.assert.ts:23]: You should test each thing being wrong separately, not all of them being wrong at the same time.

File: src/types/aggregation.assert.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    distanceMultiplier?: number
+}
+
+// Coll stats example
+ta.assert<ta.Extends<CorrectCollStatsExample, Collstats>>()
+ta.assert<ta.Not<ta.Extends<IncorrectCollStatsExample, Collstats>>>()
+
+// geoNear example
+ta.assert<ta.Equal<geoNear, Pick<GeoNear,'near'>>>()
+// ta.assert<ta.Not<ta.Equal<geoNear, Pick<GeoNear,'near'>>>()
```
Comment (tianhuil) [inline src/types/aggregation.assert.ts:43]: No commented-out code.

File: src/types/limit.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type Limit = number
```
Comment (tianhuil) [inline src/types/limit.ts:1]: These don't need their own file.

**Review by tianhuil — CHANGES_REQUESTED** [review]
Please make these changes.

File: src/types/aggregation.ts [hunk 5]
```diff
@@ -10,12 +14,48 @@ import { RemodelType } from './util'
...
   | { $lookup: TsLookup<TSchema, TSchemaOther> }
+  | {
+   /**
+   * Limits the number of documents passed to the next stage in the pipeline and takes a positive integer that specifies the maximum number of documents to pass along.
+   * 
+   * Starting in MongoDB 5.0, the $limit pipeline aggregation has a 64-bit integer limit. Values passed to the pipeline which exceed this limit will return a invalid argument error.
+   * 
+   * https://www.mongodb.com/docs/manual/reference/operator/aggregation/limit/#mongodb-pipeline-pipe.-limit
+   */ 
+    $limit: Limit
```
Comment (tianhuil) [inline src/types/aggregation.ts:31]: Just use `number`.

File: src/types/aggregation.ts [hunk 6]
```diff
@@ -10,12 +14,48 @@ import { RemodelType } from './util'
...
+  } |
+  {
+    /**
+    * Skips over the specified number of documents that pass into the stage and passes the remaining documents to the next stage in the pipeline.
+    * it takes a positive integer that specifies the maximum number of documents to skip.
+    * 
+    * Starting in MongoDB 5.0, the $skip pipeline aggregation has a 64-bit integer limit. Values passed to the pipeline which exceed this limit will return a invalid argument error.
+    * https://www.mongodb.com/docs/manual/reference/operator/aggregation/skip/#mongodb-pipeline-pipe.-skip
+    */ 
+     $skip: Skip
```
Comment (tianhuil) [inline src/types/aggregation.ts:50]: Just use `number`.

**Review by tianhuil — CHANGES_REQUESTED** [review]

File: src/types/geoNear.ts (hunk 1 above)
Comment (herberthk) [reply src/types/geoNear.ts]: Thanks @tianhuil  for your review comments, however, I could only reuse `Coordinates` from `geometry.ts` other types incompatible

Comment (herberthk) [reply src/types/geoNear.ts]: Now renamed to `TsLegacyCoordinates.ts` from `geojson.ts`

**Pull request closed without merging**